packages/frontend2/src/server/features/data-availability/utils/get-da-projects-tvl.ts

```diff
@@ -0,0 +1,56 @@
+import { ProjectId } from '@l2beat/shared-pure'
+import { groupBy } from 'lodash'
+import {
+  unstable_cache as cache,
+  unstable_noStore as noStore,
+} from 'next/cache'
+import { db } from '~/server/database'
+
+export async function getDaProjectsTvl(projectIds: ProjectId[]) {
+  noStore()
+  return await getCachedDaProjectsTvl(projectIds)
+}
+
+const getCachedDaProjectsTvl = cache(async (projectIds: ProjectId[]) => {
+  const values = await db.value.getLatestValuesForProjects(projectIds)
+
+  const byProject = groupBy(values, 'projectId')
+
+  const aggregated = Object.entries(byProject).map(([projectId, values]) => {
+    const { canonical, external, native } = values.reduce(
+      (acc, value) => {
+        acc.canonical += value.canonical
+        acc.external += value.external
+        acc.native += value.native
+
+        return acc
+      },
+      { canonical: 0n, external: 0n, native: 0n },
+    )
+
+    const tvl = canonical + external + native
+
+    return {
+      projectId: ProjectId(projectId),
+      tvl: Number(tvl),
```

Comment:
is it safe to cast it to a number here?

Reply:
Yea I would say so given:
- tvl is denominated in fiat/USD cents:
- max safe int here is around 9 quadrillion